thoth/lab/inspection_report.py

```diff
@@ -23,234 +23,96 @@
-            )
-            inspection_report[feature] = inspection.show_categories(feature_result)
-
-    return inspection_report
-
-
-def create_inspection_reports(inspection_df_dict: dict) -> dict:
-    """Create dictionary containing all reports for inspection batches selected.
-
-    :param inspection_df_dict: dictionary with inspection_df per identifier as returned by process_inspection_results
-    """
-    identifier_list = list(inspection_df_dict.keys())
-
-    inspection_report_dict = {}
-
-    for identifier in identifier_list:
-        inspection_report_dict[identifier] = create_inspection_report(inspection_df_dict[identifier])
-
-    return inspection_report_dict
-
-
-def create_feature_analysis_summary(inspection_report_dict: dict, explanation: bool = False):
-    """Create summary of analysis of features across all inspection reports.
-
-    :param inspection_report_dict: dictionary of inspection report as returned by create_inspection_report_dict
-    :param explanation: flag to obtain more detailed results about differences in the inspection batches
-    """
-    results_features = _aggregate_results_per_feature(inspection_report_dict)
-    features_summary = {}
-
-    if explanation:
-        explanation_summary = {}
-
-    for feature, feature_results in results_features.items():
-        if explanation:
-            explanation_summary[feature] = {}
-
-        if not isinstance(feature_results, list):
-            features_summary[feature] = {}
-
-            for sub_feature, sub_feature_results in feature_results.items():
-                if explanation:
-                    explanation_summary[feature][sub_feature] = {}
-                keys = [key for key in sub_feature_results[0].keys()]
-                key_counts = {}
-
-                for key in keys:
-
-                    key_counts[key] = len(set([k[key] for k in results_features[feature][sub_feature]]))
-                    if explanation:
-                        if key_counts[key] != 1:
-                            explanation_summary[feature][sub_feature][key] = set(
-                                [k[key] for k in results_features[feature][sub_feature]]
-                            )
-
-                features_summary[feature][sub_feature] = key_counts
+def multi_table(table_dict):
+    ''' Acceps a list of IpyTable objects and returns a table which contains each IpyTable in a cell
+    '''
+    return HTML(
+        '<table><br style="background-color:white;">' + 
+        ''.join(['<br>' + table._repr_html_() + '</br>' for table in table_dict.values()]) +
+        '</br></table>'
+    )
+
+def create_df_report(df: pd.DataFrame) -> pd.DataFrame:
+    """Show unique values for each column in the dataframe."""
+    dataframe_report = {}
+    for c_name in df.columns.values:
+        try:
+            unique_values = df[c_name].unique()
+            dataframe_report[c_name] = [unique_values] 
+        except Exception as e:
```

Comment:
What kind of exception is thrown here?

Reply:
I'll create exception file for thoth-lab